examples/sudoku.coffee

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env coffee
+try
+  FD = require '../src/index' # running from source-tree
+catch e
+  FD = require 'finitedomain'
+
+# Setup rules that holds for all Sudoku games
+# https://en.wikipedia.org/wiki/Sudoku
+rows = ['A','B','C','D','E','F','G','H','I']
+cols = ['1','2','3','4','5','6','7','8','9']
+setupSudoku = (S) ->
+  vars = []
+
+  # Declare possible board places
+  for i in [0...9]
+    for j in [0...9]
+      varName = rows[i] + cols[j]
+      vars.push varName
+      S.addVar varName, [1,9]
+
+  # Add row constraints.
+  for i in [0...9]
+    k = []
+    for j in [0...9]
+        k.push(rows[i] + cols[j])
+    S.distinct k
+
+  # Add column constraints
+  for i in [0...9]
+    k = []
+    for j in [0...9]
+      k.push(rows[j] + cols[i])
+    S.distinct k
+
+  # Add box constraints.
+  for i in [0...3]
+    for j in [0...3]
+      k = []
+      for i2 in [0...3]
+        for j2 in [0...3]
+          k.push(rows[i * 3 + i2] + cols[j * 3 + j2])
+      S.distinct k
+
+  return vars
+
+# Parses Sudoku puzzles on format from
+# http://www2.warwick.ac.uk/fac/sci/moac/people/students/peter_cock/python/sudoku
+parseBoard = (vars, str) ->
+  board = {}
+  for char, index in str
+    position = vars[index]
+    board[position] = parseInt(char) if char != '.'
```

Comment:
We don't use statement modifiers in DS (the trailing `if`).
Nitpick because I know it's not relevant here, but `parseInt` takes a radix argument and it's best to just always use it to prevent accidentally forgetting about it when it does matter.